geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegionMap.java

```diff
@@ -388,9 +378,21 @@ public void close(BucketRegion bucketRegion) {
     if (logger.isDebugEnabled()) {
       logger.debug("Clearing entries for {} rvv={}", _getOwner(), rvv);
     }
-    LocalRegion lr = _getOwner();
+    final LocalRegion lr = _getOwner();
     RegionVersionVector localRvv = lr.getVersionVector();
-    incClearCount(lr);
+
+    boolean isBucketRegion = false;
+
+    if (!(lr instanceof HARegion)) {
+      CachePerfStats stats = lr.getCachePerfStats();
+      if (stats != null) {
+        if (lr.isUsedForPartitionedRegionBucket()) {
```

Comment:
this code would be much cleaner if you added noop methods on LocalRegion named startBucketClear and endBucketClear. Then you can override those methods on BucketRegion to call the corresponding start/end on the prstats. That allows you to get rid of the ifs and the isBucketRegion boolean here.

Reply:
Done.